control-plane/pkg/reconciler/source/source.go

```diff
@@ -185,6 +185,10 @@ func (r *Reconciler) getResource(ctx context.Context, ks *sources.KafkaSource, s
 		EgressConfig:  egressConfig,
 		DeliveryOrder: DefaultDeliveryOrder,
 	}
+	// Set key type hint (if any).
+	if keyType, ok := ks.Labels[sources.KafkaKeyTypeLabel]; ok {
```

Comment:
I am open to making this change, but why not a proper spec field?
```
spec:
  key:
    type: <int | string | byte-array | float>
```

A spec field has many benefits over an annotation like validation since if I mistype the annotation key or label key I won't get any error or warning.

Reply:
Oh, that's an interesting point, making this part of the spec 

I don't actually know why... I guess less "heavy" to do as label/annotation.

But sounds reasonable to me...  but we should keep the existing `label`, and just at some point "migrate" 

At least my $0.02